Refactor deployContexts to StringKeyMap configContexts

paul-thompson-helix opened the pull request:
* Rename  `DeployContextSource` to `ConfigContextSource`

* Define and use `StringKeyMap` in ADL definitions instead of `StringMap` in order to name the keys , Including on maps in Config and State.

* Add new `StringKeyMap<StaticConfigTopicName, ConfigContextSource> configContexts = {};`  to replace `Vector<DeployContext> deployContexts` where `DeployContext = Struct { String name; ConfigContextSource; }`

* Drop config.adl `DeployContext`
* Rename DeployConfigTopicName to StaticConfigTopicName
* Rename fetchDeployContext to fetchConfigContext

* Drop backwards compatibility for original `Toolconfig.deployContexts` field

paul-thompson-helix committed:
Use StringKeyMap in ADL definitions of StringMap including naming the keys.

paul-thompson-helix committed:
Propagate name change DeployContextSource -> ConfigContextSource

paul-thompson-helix committed:
Drop backwards compatibility for ToolConfig.deployContexts

Drop config.adl DeployContext
Rename DeployConfigTopicName to StaticConfigTopicName
Rename fetchDeployContext to fetchConfigContext

paul-thompson-helix commented:
Extracted from work in progress on dynamic config.  

NB: Requires testing.

paul-thompson-helix commented:
Rebasing ahead of current origin/master

timbod7 commented on adl/config.adl:
```diff
@@ -23,7 +25,9 @@ struct ToolConfig {
 
-    Vector<DeployContext> deployContexts = [];
+    /// Static Config contexts
+    StringKeyMap<StaticConfigTopicName, ConfigContextSource> configContexts = {};
```
A couple of suggested renames: 

    - StaticConfigTopicName => StaticConfigName
    - ConfigContextSource => JsonSource

paul-thompson-helix committed:
Rename to StringKeyMap<StaticConfigName, JsonSource> configSources = {};

paul-thompson-helix committed:
Copy built binary to ./tests

paul-thompson-helix committed:
self contained demo of hx-deploy-tool & a basic config templating

creation of a zip release
uploading bits of config to s3
setting up configs in the right places
executing hx-deploy-tool start release-xx-yy-zz.zip

paul-thompson-helix commented:
Added a self contained demo of hx-deploy-tool & a basic config templating example.

Useful to test this branch & for doco generally
